src/features/ProfileView/Profile.History.js

```diff
@@ -44,26 +41,56 @@ const ProfileHistory = () => {
     {restauraunt: 'Taqueria El Rey', hostStatus: 'Host', bill: 8.49}
   ];
 
+  const [biteHistory, setBiteHistory] = useState(mockBites);
+  const { state: { userId }, dispatch } = useContext(BiteShareContext);
+
   const renderBite = ({item, index}) => (<PreviousBite meal={item} index={index}/>);
 
-  //TODO
-  //Query previous bites from db
-  //Restauraunt Name - Guest / Host Status - Price
-  //Place in array in state to be rendered via Flatlist
+  useEffect(() => {
+    getAllDocuments(`users/${userId}/transactions`)
+      .then((user) => {
+        let cleanedTransactions = biteHistory;
+
+        user.forEach((session) => {
+          readASingleDocument(`users/${userId}/transactions`, session.id)
+            .then((summary) => {
+              summary = summary.data();
+              let bite = {
+                restauraunt: summary.restaurantName,
+                bill: summary.individualBills,
+                hostStatus: summary.role || 'guest'
+              };
+              cleanedTransactions.push(bite);
+            })
+            .catch((err) => {
+              console.log('Error pulling transaction history for meal session: ', session.id);
+            });
+        });
+
+        setBiteHistory(cleanedTransactions);
+      })
+      .catch((err) => {
+        console.log('Error pulling transaction history for user: ', userId );
+      });
+
+  }, []);
```

Comment:
```suggestion
  useEffect(() => {
    readCollectionSnapshotListener(`users/${userId}/transactions`, (transactions) => {
      let cleanedTransactions = [...biteHistory];
      transactions.forEach((session) => {
        const { restaurantName, individualBills, role } = session.data();
        const bite = {
          restaurant: restaurantName,
          bill: individualBills,
          hostStatus: role || 'Guest'
        };
        cleanedTransactions.unshift(bite);
      });
      setBiteHistory(cleanedTransactions);
    })
  }, []);
```

Reply:
Love this suggestion. Didn't realize that fx() could be used to access all docs directly - reduces nesting and is way cleaner. I'll make the changes on my home machine after work today.